browser/speedreader/speedreader_tab_helper.h

```diff
@@ -0,0 +1,46 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_BROWSER_SPEEDREADER_SPEEDREADER_TAB_HELPER_H_
+#define BRAVE_BROWSER_SPEEDREADER_SPEEDREADER_TAB_HELPER_H_
+
+#include "content/public/browser/web_contents_observer.h"
+#include "content/public/browser/web_contents_user_data.h"
+
+namespace content {
+class WebContents;
+}
+
+namespace speedreader {
+
+// Updates the speedreader toolbar button state when a tab visibility changes.
+class SpeedreaderTabHelper
```

Comment:
Whenever browser ui change needs, `BraveToolBarView::Update(content::WebContents*)` is called.
I think we can manage speedreader button's state at there w/o this tab helper.
Also, using views api outside of views modules is layer violation.

Reply:
I don't think the toolbar should be aware of speedreader button activation logic, so I propose to keep the helper.
Agree that there is a violation, I'm open for suggestions :) We probably should involve the `Browser` object, as bookmarks do

Reply:
I mean we can call `SpeedReaderButton::Update` from `ToolBarView`.
Maybe speedreader button can have the logic that tab helper has?

Reply:
With tab helper, I think it's hard to remove this layer violation.

Reply:
The main problem is that the "active" state is determined in BraveContentBrowserClient - we should pass it to the toolbar/button somehow. So it seems the only way is to user `Browser` - i.e. subclass it, add a method and override it in `BraveBrowserView`

Reply:
Ok - I see. How about this?
Setting active state to tab helper from here and checking that state from SpeedReaderButton::Update()?
With this, I think we don't need to introduce additional api to BrowserView.

Reply:
I did a small refactor, so now Tab Helper determines the active state by itself and the toolbar actively queries this state - PTAL :)